src/test/java/org/jenkinsci/plugins/workflow/DSLTest.java

```diff
@@ -56,4 +57,74 @@
         r.assertLogContains("What is the message?", r.assertBuildStatusSuccess(p.scheduleBuild2(0)));
     }
 
+    /**
+     * Tests the ability to execute meta-step with clean syntax
+     */
+    @Issue("JENKINS-29922")
+    @Test
+    public void dollar_class_must_die() throws Exception {
+        WorkflowJob p = r.jenkins.createProject(WorkflowJob.class, "die1");
+        p.setDefinition(new CpsFlowDefinition("california ocean:'pacific', mountain:'sierra'"));
+        r.assertLogContains("California from pacific to sierra", r.assertBuildStatusSuccess(p.scheduleBuild2(0)));
+    }
+
+    /**
+     * Split arguments between meta step and state
+     */
+    @Issue("JENKINS-29922")
+    @Test
+    public void dollar_class_must_die2() throws Exception {
+        WorkflowJob p = r.jenkins.createProject(WorkflowJob.class, "die2");
+        p.setDefinition(new CpsFlowDefinition("california ocean:'pacific', mountain:'sierra', moderate:true"));
+        r.assertLogContains("Introducing california\nCalifornia from pacific to sierra", r.assertBuildStatusSuccess(p.scheduleBuild2(0)));
+    }
+
+    /**
+     * Split arguments between meta step and state
+     */
+    @Issue("JENKINS-29922")
+    @Test
+    public void dollar_class_must_die3() throws Exception {
+        WorkflowJob p = r.jenkins.createProject(WorkflowJob.class, "die3");
+        p.setDefinition(new CpsFlowDefinition("nevada()"));
+        r.assertLogContains("All For Our Country", r.assertBuildStatusSuccess(p.scheduleBuild2(0)));
+    }
+
+    /**
+     * Split arguments between meta step and state, when argument is colliding
+     */
+    @Issue("JENKINS-29922")
+    @Test
+    public void dollar_class_must_die_colliding_argument() throws Exception {
+        WorkflowJob p = r.jenkins.createProject(WorkflowJob.class, "die5");
+        p.setDefinition(new CpsFlowDefinition("newYork motto:'Empire', moderate:true"));
+        WorkflowRun run = r.assertBuildStatusSuccess(p.scheduleBuild2(0));
+        r.assertLogContains("Introducing newYork\nThe Empire State", run);
+        r.assertLogNotContains("New York can be moderate in spring or fall", run);
+    }
+
+    /**
+     * Single argument state
+     */
+    @Issue("JENKINS-29922")
+    @Test
+    public void dollar_class_must_die_onearg() throws Exception {
+        WorkflowJob p = r.jenkins.createProject(WorkflowJob.class, "die4");
+        p.setDefinition(new CpsFlowDefinition("newYork 'Empire'"));
+        r.assertLogContains("The Empire State", r.assertBuildStatusSuccess(p.scheduleBuild2(0)));
+    }
+
+    @Issue("JENKINS-29922")
+    @Test public void runMetaBlockStep() throws Exception {
```

Comment:
Would still like to see tests/how this would look for `properties`, but that's probably fairly special-casing and would likely be in `workflow-multibranch-plugin` anyway.